### ISSUE TITLE
Cancel/Finish NavigationView in Fragment

### ISSUE THREAD

**Issue by clinkadink** [mapbox/mapbox-navigation-android#979]
I am using the latest public release and cannot cancel or finish navigation (NavigationView) inside a fragment. I have one main activity, a container and several fragments. I launch the navigation view fragment from a map view (MapView) fragment. When navigation is running and I am in the navigation view (NavigationView) fragment, I can go back to the the map view fragment - and the navigation view is still running.

MainActivity
   |__ FragMap
   |__ FragNav

**Android API:** 27
**Mapbox Navigation SDK version:** 0.13.0

```
implementation('com.mapbox.mapboxsdk:mapbox-android-sdk:6.1.3@aar') {
        transitive = true
}
implementation 'com.mapbox.mapboxsdk:mapbox-android-navigation:0.13.0'
implementation('com.mapbox.mapboxsdk:mapbox-android-navigation-ui:0.13.0') {
        transitive = true
}
```
In the NavigationView fragment:
```
    @Override
    public void onCancelNavigation() {
        if (getActivity() != null) {
            getActivity().finish();
        }
    }
```
### Steps to trigger behavior

 1. Run the app and initialise the navigation view
 2. Either let the navigation finish, or cancel the navigation (using the X)
 3. The app immediately crashes without any logcat errors

### Expected behavior
I would expect the navigation to end/stop/cancel, and return to the previous fragment (in backstack), i.e. my map view fragment.

### Actual behavior
When pressing the 'X' button in the navigation view the whole app crashes.

**Note 1:** I did try updating to the latest snapshot (0.14.0) for both navigation and navigation-ui, but several methods wouldn't validate. E.g. NavigationViewOptions had about half the options missing, such as origin, unit type, etc. So I had to revert to 0.13.0.

**Note 2:** I have tried swapping getActivity().finish() for navigationView.finishNavigationView(), but it doesn't seem to do anything. Although the app doesn't crash.

**Note 3:** my code for the navigation view fragment is exactly as documented [here](https://www.mapbox.com/android-docs/navigation/overview/navigation-ui/#navigationview-fragment-example).

I have already seen that this has been raised [in #813](https://github.com/mapbox/mapbox-navigation-android/issues/813).

Finally, thanks for all the hard work that has gone into this SDK. It is to say the least ... impressive.

**Comment by danesfeder** [comment]
@clinkadink Yeah this wasn't you doing anything wrong - it is us breaking the API.  Thank you, I will take a look at your code and try to see what's going on here.  Stayed tuned for `0.14.0`, we are working quickly to get it out this week.

**Comment by danesfeder** [comment]
Tracking with #959

**Comment by Guardiola31337** [comment]
Hey @clinkadink 👋 this should be fixed with #959 which landed last week and was included in the latest release [`0.16.0-beta.1`](https://github.com/mapbox/mapbox-navigation-android/releases/tag/v0.16.0-beta.1). Please feel free to re-open if you still experience this issue.

Thanks for bringing this to us 🙏
